### PR TITLE
Drop unused GetType() from CSizeComputer

### PR DESCRIPTION
**Pull request by Empact**
Based on conversation in #13462, it seems the serialization `GetType` has very narrow use/effect. In every case except for `CAddress`, which specifically relates to a network peer's address, not a wallet address etc., the serialized representation of an object is irrespective of its destination / type.

This removes the unused `GetType` method from `CSizeComputer` as a step to further narrowing that use.

### CODE REVIEW

**Comment by sipa** [comment]
utACK 60cf69ee7855b8d5c86d3ed41c47f764ccee5095. Nice simplification.

**Comment by laanwj** [comment]
I wonder if this is slightly confusing as long as long as it still uses the type parameter in the rest of (de)serialization.

Long term I do think that it makes sense to get rid of it altogether. Instead of passing a type into the serialization it'd be more clear to use a subclass or proxy object.

**Comment by Empact** [comment]
Rebased for #13792

File: src/serialize.h [hunk 1]
```diff
@@ -901,10 +901,9 @@ class CSizeComputer
 protected:
     size_t nSize;
 
-    const int nType;
     const int nVersion;
 public:
-    CSizeComputer(int nTypeIn, int nVersionIn) : nSize(0), nType(nTypeIn), nVersion(nVersionIn) {}
+    CSizeComputer(int nVersionIn) : nSize(0), nVersion(nVersionIn) {}
```
Comment (practicalswift) [inline src/serialize.h]: Should be `explicit`? :-)

Comment (laanwj) [reply src/serialize.h]: yes, that's good practice for one-argument constructors

**Comment by Empact** [comment]
Made the now single-arg constructor `explicit`

**Comment by kallewoof** [comment]
@Empact Can you give hints on how to fix the below? Code no longer compiles..

```C++
    template <typename Stream, typename Operation>
    inline void SerializationOp(Stream& s, Operation ser_action) {
        READWRITE(this->nVersion);
        READWRITE(hashPrevBlock);
        READWRITE(hashMerkleRoot);
        READWRITE(nTime);
        READWRITE(nBits);
        READWRITE(nNonce);
        if (g_solution_blocks && !(s.GetType() & SER_GETHASH)) {
            READWRITE(g_blockheader_payload_map[GetHash()]);
            size_t len = GetSizeOfCompactSize(g_blockheader_payload_map[GetHash()].size()) + g_blockheader_payload_map[GetHash()].size();
            while (len < g_solution_block_len) {
                uint8_t padding = 0;
                READWRITE(padding);
                len++;
            }
        }
    }
```

**Pull request merged**